chainer/distributions/independent.py

```diff
@@ -60,10 +58,12 @@ def batch_shape(self):
     def event_shape(self):
         return self.__event_shape
 
-    @property
+    @cache.cached_property
     def covariance(self):
-        '''Returns the covariance of the distribution based on the original
-        i.i.d. distribution. By definition, the covariance of the new
+        """ The covariance of the distribution based on the original
+        i.i.d. distribution.
```

Comment:
Could you make the summary shorter to follow H405?

> [H405] Multi line docstrings should start with a one line summary followed by an empty line.

https://docs.openstack.org/hacking/latest/user/hacking.html#docstrings

In the previous review, I was going to include a comment like this on the line.  I'm sorry that I didn't.

Reply:
I missed that, sorry. Thank you for pointing out. I fixed them in cbc0e2b